spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala

```diff
@@ -17,86 +17,95 @@
 package com.intel.analytics.bigdl.dataset
 
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.bigdl.tensor.{DoubleType, FloatType, Storage, Tensor}
+import com.intel.analytics.bigdl.tensor.Tensor
+import org.apache.commons.lang3.SerializationUtils
 
 import scala.reflect.ClassTag
 
 /**
- * Sample, bundling input and target
- *
- * @param featureTensor
- * @param labelTensor
- * @tparam T
+ * Class that represents the features and labels of a data sample.
+ * @tparam T numeric type
  */
+abstract class Sample[T: ClassTag] extends Serializable {
+  /**
+   * Number of elements in features and labels.
+   * This function will be used in sorting samples.
+   * @return
+   */
+  def nElement(): (Int, Int)
```

Comment:
Implement a featureLength() method instead - see the comments above (in sortData)